part_1/solution.py

```diff
@@ -0,0 +1,1327 @@
+    )
+
+# %%
+# Create log directory if needed, and launch tensorboard
+log_dir.mkdir(parents=True, exist_ok=True)
+
+# %% [markdown] tags=[]
+"""
+The next cell starts tensorboard.
+
+<div class="alert alert-warning">
+If you launched jupyter lab from ssh terminal, add <code>--host &lt;your-server-name&gt;</code> to the tensorboard command below. <code>&lt;your-server-name&gt;</code> is the address of your compute node that ends in amazonaws.com.
+
+</div>
+
+<div class="alert alert-warning">
+If you are using VSCode and a remote server, you will need to forward the port to view the tensorboard. <br>
+Take note of the port number was assigned in the previous cell.(i.e <code> http://localhost:{port_number_assigned}</code>) <br>
+
+Locate the your VSCode terminal and select the <code>Ports</code> tab <br>
+<ul>
+<li>Add a new port with the <code>port_number_assigned</code>
+</ul>
+Click on the link to view the tensorboard and it should open in your browser.
+</div>
+"""
+
+# %% Imports and paths tags=[]
+
+
+# Function to find an available port
+def find_free_port():
+    import socket
+
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("", 0))
+        return s.getsockname()[1]
+
+
+# Launch TensorBoard on the browser
+def launch_tensorboard(log_dir):
+    import subprocess
+
+    port = find_free_port()
+    tensorboard_cmd = f"tensorboard --logdir={log_dir} --port={port}"
+    process = subprocess.Popen(tensorboard_cmd, shell=True)
+    print(
+        f"TensorBoard started at http://localhost:{port}. \n"
+        "If you are using VSCode remote session, forward the port using the PORTS tab next to TERMINAL."
+    )
+    return process
+
+
+# Launch tensorboard and click on the link to view the logs.
+tensorboard_process = launch_tensorboard(log_dir)
+
+# %% [markdown]
+"""
+## Load OME-Zarr Dataset
+
+There should be 34 FOVs in the dataset.
+
+Each FOV consists of 3 channels of 2048x2048 images,
+saved in the [High-Content Screening (HCS) layout](https://ngff.openmicroscopy.org/latest/#hcs-layout)
+specified by the Open Microscopy Environment Next Generation File Format
+(OME-NGFF).
+
+- The layout on the disk is: `row/col/field/pyramid_level/timepoint/channel/z/y/x.`
+- These datasets only have 1 level in the pyramid (highest resolution) which is '0'.
+"""
+
+# %% [markdown]
+"""
+<div class="alert alert-warning">
+You can inspect the tree structure by using your terminal:
+<code> iohub info -v "path-to-ome-zarr" </code>
+
+<br>
+More info on the CLI:
+<code>iohub info --help </code> to see the help menu.
+</div>
+"""
+# %%
+# This is the python function called by `iohub info` CLI command
+print_info(data_path, verbose=True)
+
+# Open and inspect the dataset.
+dataset = open_ome_zarr(data_path)
+
+# %%
+# Use the field and pyramid_level below to visualize data.
+row = 0
+col = 0
+field = 9  # TODO: Change this to explore data.
```

Comment:
If you really want them to do this you should make it an explicit todo, else some people may just skip over